core/services/llo/channeldefinitions/onchain_channel_definition_cache.go

```diff
@@ -192,11 +241,87 @@ func (c *channelDefinitionCache) Start(ctx context.Context) error {
 	})
 }
 
-////////////////////////////////////////////////////////////////////
-// Log Polling
-////////////////////////////////////////////////////////////////////
+// blockNumFromUint64 converts a uint64 block number to int64
+// This is safe as block numbers are well within int64 range
+func blockNumFromUint64(blockNum uint64) int64 {
+	//nolint:gosec // disable G115
+	return int64(blockNum)
+}
+
+// unpackOwnerLog unpacks and validates an owner log from logpoller
+// Returns the unpacked log and an error if unpacking or validation fails
+func (c *channelDefinitionCache) unpackOwnerLog(log logpoller.Log) (*channel_config_store.ChannelConfigStoreNewChannelDefinition, error) {
+	if log.EventSig != NewChannelDefinition {
+		return nil, fmt.Errorf("log event signature mismatch: expected %x, got %x", NewChannelDefinition, log.EventSig)
+	}
+
+	unpacked := new(channel_config_store.ChannelConfigStoreNewChannelDefinition)
+	err := channelConfigStoreABI.UnpackIntoInterface(unpacked, newChannelDefinitionEventName, log.Data)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unpack log data: %w", err)
+	}
+
+	if len(log.Topics) < 2 {
+		return nil, fmt.Errorf("log missing expected topics: got %d, expected at least 2", len(log.Topics))
+	}
+
+	unpacked.DonId = new(big.Int).SetBytes(log.Topics[1])
+	//nolint:gosec // disable G115
+	unpacked.Raw.BlockNumber = uint64(log.BlockNumber)
+
+	// Validate donID matches
+	if unpacked.DonId.Cmp(big.NewInt(int64(c.donID))) != 0 {
+		return nil, fmt.Errorf("donID mismatch: expected %d, got %s", c.donID, unpacked.DonId.String())
+	}
+
+	return unpacked, nil
+}
+
+// unpackAdderLog unpacks and validates an adder log from logpoller
+// Returns the unpacked log and an error if unpacking or validation fails
+func (c *channelDefinitionCache) unpackAdderLog(log logpoller.Log) (*channel_config_store.ChannelConfigStoreChannelDefinitionAdded, error) {
```

Comment:
same comment - wherever the new bindings are should be the function `ParseChannelDefinitionAdded`